mitmproxy/proxy/protocol/http2.py

```diff
@@ -487,9 +487,17 @@ def read_request_headers(self, flow):
 
     @detect_zombie_stream
     def read_request_body(self, request):
-        self.request_data_finished.wait()
```

Comment:
We probably want to keep the wait for non-streaming requests (as you had before)